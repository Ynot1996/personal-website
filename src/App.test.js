import React from 'react';
import { render } from '@testing-library/react';
import GlobalStyle from './App'; // 假設 GlobalStyle 定義在 App 中

describe('GlobalStyle', () => {
  test('applies global styles correctly', () => {
    const { container } = render(
      <>
        <GlobalStyle />
        <div data-testid="test-element">Test</div>
      </>
    );

    // 確保樣式正確應用
    expect(container.firstChild).toMatchSnapshot();
  });
});

// 僅針對特定警告靜音
beforeAll(() => {
  jest.spyOn(console, 'error').mockImplementation((message) => {
    if (
      message.includes('React does not recognize the') || // React 的 DOM 屬性警告
      message.includes('React Router Future Flag Warning') // React Router 的未來相容性提醒
    ) {
      return; // 靜音這些已知警告
    }
    console.error(message); // 保留其他錯誤
  });

  jest.spyOn(console, 'warn').mockImplementation((message) => {
    if (
      message.includes('Relative route resolution') // React Router 路由相容性警告
    ) {
      return; // 靜音特定警告
    }
    console.warn(message); // 保留其他警告
  });
});

afterAll(() => {
  console.error.mockRestore();
  console.warn.mockRestore();
});
