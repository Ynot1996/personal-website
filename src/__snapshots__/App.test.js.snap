// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle applies global styles correctly 1`] = `
<nav
  class="sc-KXCwU bfqIkq"
>
  <a
    class="sc-gyycJP lOWGQ"
    href="/personal-website"
  >
    <img
      alt="Logo"
      class="sc-gaZyOd hwMeGZ"
      src="/assets/images/logo.png"
    />
  </a>
  <div
    class="sc-kuACkN gDAsVo"
  >
    <div
      class="sc-hWWBcw sc-jHbxoU fcfWZC dOAYdY"
    />
    <div
      class="sc-hWWBcw sc-cGXZpB fcfWZC iyfwzc"
    />
    <div
      class="sc-hWWBcw sc-jZCRgm fcfWZC eZmUKx"
    />
  </div>
  <ul
    class="sc-dDvxFM bgbFEa"
  >
    <a
      class="sc-eaUbBy ebNQtv"
      href="/personal-website"
    >
      <img
        alt="Logo"
        class="sc-gaZyOd hwMeGZ"
        src="/assets/images/logo.png"
      />
    </a>
    <li
      class="sc-jqVXSH gpbNph"
    >
      <a
        class="sc-gTrWKq gGkAYA ({
            isActive
          }) => isActive ? \\"active\\" : \\"\\""
        href="/personal-website"
      >
        Home
      </a>
    </li>
    <li
      class="sc-jqVXSH gpbNph"
    >
      <a
        class="sc-gTrWKq gGkAYA ({
            isActive
          }) => isActive ? \\"active\\" : \\"\\""
        href="/personal-website/about"
      >
        About
      </a>
    </li>
    <li
      class="sc-jqVXSH gpbNph"
    >
      <a
        class="sc-gTrWKq gGkAYA ({
            isActive
          }) => isActive ? \\"active\\" : \\"\\""
        href="/personal-website/projects"
      >
        Projects
      </a>
    </li>
    <li
      class="sc-jqVXSH gpbNph"
    >
      <a
        class="sc-gTrWKq gGkAYA ({
            isActive
          }) => isActive ? \\"active\\" : \\"\\""
        href="/personal-website/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
